synonymdev/react-native-ldk
# feat(lightning): Watch Transactions & Outputs

This PR:
- Adds three additional methods to the lightning manager start param `getScriptPubKeyHistory`, `getAddress` & `broadcastTransaction`. 
  - `getScriptPubKeyHistory` is used to check the current status of the provided watch transactions and outputs. 
  - `getAddress` is used to specify an address to send sats to when closing channels.
  - `broadcastTransaction` is used to broadcast a rawTx for the sweeping of funds from a channel closure.
  - With this, we are now able to close a given channel and sweep to a specified address when ready.
- Adds `confirmed_outputs`, `confirmed_transactions` & `broadcastedTransactions` to `ELdkData`, `TLdkData`, `ELdkFiles` & `DefaultLdkDataShape`.
- Saves and checks for duplicate watch transactions, outputs and broadcasted transactions.
- Adds `vout` to `getTransactionData` response.
- Updates `syncLdk`, `onChannelManagerSpendableOutputs`, `onRegisterTx`, `onRegisterOutput` & `onBroadcastTransaction` methods in `lightning-manager.ts`.
- Adds `bitcoinjs-lib` as a lib dependency.
- Updates the example app accordingly.
- Removes outdated example from `README.md`.
- Bumps version to `0.0.55`.

To Test:
- Using the example app, open a lightning channel containing a balance on each side.
- Tap, "Close channel" and choose to close cooperatively ("Close channel").
- Mine 1 block and then 6 more.
- Tap "Get Address Balance"
- You should see a positive balance containing the closed channel's swept funds.
